### PR TITLE
Remove PlayerPreprocessCommandEvent listener and clean up Commands

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

Skript currently listens to PPCE to check if a player is running a custom Skript command. If it determines they are, it cancels the event and runs the command. This, first of all, is a misuse of the Bukkit API. Secondly, it causes issues when listening with `on command` at priority HIGHEST or MONITOR. Thirdly, it prevents other plugins from being able to cancel or otherwise manipulate custom Skript commands.

Skript already registers commands properly via the Bukkit API, and has a working onCommand listener in the ScriptCommand class. This PR removes the PPCE listener and re-routes the commands through their proper onCommand listeners. It also renames the `log player commands` config option to `log effect commands`, as it no longer affects the logging of custom Skript commands, and hasn't for a while, I believe.

Finally, some other miscellaneous cleanup was done to Commands.java.

This PR needs more testing. I've done some and it all seems fine, but I'd be more comfortable if team members or Skript users could confirm that this doesn't break any existing behavior.

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:** #5752 #5908

### CODE REVIEW

**Review by Moderocky — APPROVED** [review]

**Comment by AyhamAl-Ali** [comment]
Can be sent to testing discord for more testing

File: src/main/java/ch/njol/skript/command/Commands.java [hunk 1]
```diff
@@ -137,64 +134,27 @@ private static void init() {
...
+		public void onServerCommand(final ServerCommandEvent event) {
+			if (event.getCommand().isEmpty() || event.isCancelled())
 				return;
-			if ((Skript.testing() || SkriptConfig.enableEffectCommands.value()) && e.getCommand().startsWith(SkriptConfig.effectCommandToken.value())) {
-				if (handleEffectCommand(e.getSender(), e.getCommand())) {
-					e.setCancelled(true);
+			if ((Skript.testing() || SkriptConfig.enableEffectCommands.value()) && event.getCommand().startsWith(SkriptConfig.effectCommandToken.value())) {
+				if (handleEffectCommand(event.getSender(), event.getCommand())) {
+					event.setCancelled(true);
 				}
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/command/Commands.java:154]:
```suggestion
				if (handleEffectCommand(event.getSender(), event.getCommand()))
					event.setCancelled(true);
```

File: src/main/java/ch/njol/skript/command/Commands.java [hunk 2]
```diff
@@ -303,30 +263,25 @@ public static void registerListeners() {
 
 			Bukkit.getPluginManager().registerEvents(new Listener() {
 				@EventHandler(priority = EventPriority.LOWEST, ignoreCancelled = true)
-				public void onPlayerChat(final AsyncPlayerChatEvent e) {
-					if (!SkriptConfig.enableEffectCommands.value() || !e.getMessage().startsWith(SkriptConfig.effectCommandToken.value()))
+				public void onPlayerChat(final AsyncPlayerChatEvent event) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/command/Commands.java:266]:
```suggestion
				public void onPlayerChat(AsyncPlayerChatEvent event) {
```

File: src/main/java/ch/njol/skript/command/Commands.java [hunk 3]
```diff
@@ -303,30 +263,25 @@ public static void registerListeners() {
...
-										e.setCancelled(true);
+										event.setCancelled(true);
 									break;
-								} catch (final InterruptedException e1) {
+								} catch (final InterruptedException ignored) {
 								}
 							}
-						} catch (final ExecutionException e1) {
-							Skript.exception(e1);
+						} catch (final ExecutionException e) {
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/command/Commands.java:283]:
```suggestion
						} catch (ExecutionException e) {
```

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsSkriptCommand.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
  */
 package ch.njol.skript.conditions;
 
-import static ch.njol.skript.command.Commands.skriptCommandExists;
+import static ch.njol.skript.command.Commands.scriptCommandExists;
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondIsSkriptCommand.java:21]: I'm not sure there's a reason for this to be a static import considering it's used only once. We may want to consider changing it while we're here :)

File: src/main/java/ch/njol/skript/command/Commands.java [hunk 5]
```diff
@@ -303,30 +271,25 @@ public static void registerListeners() {
...
+						if (handleEffectCommand(event.getPlayer(), event.getMessage()))
+							event.setCancelled(true);
 					} else {
-						final Future<Boolean> f = Bukkit.getScheduler().callSyncMethod(Skript.getInstance(), new Callable<Boolean>() {
-							@Override
-							public Boolean call() throws Exception {
-								return handleEffectCommand(e.getPlayer(), e.getMessage());
-							}
-						});
+						Future<Boolean> f = Bukkit.getScheduler().callSyncMethod(Skript.getInstance(), () -> handleEffectCommand(event.getPlayer(), event.getMessage()));
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/command/Commands.java:281]: This seems to replicate the behavior of the Task.callSync method. Might be worth using it here?

**Review by APickledWalrus — APPROVED** [review]
Looks good. I have a couple non-blocking suggestions.

**Pull request merged**